lib/components/os/resinos.js

```diff
@@ -71,3 +67,29 @@ exports.injectNetworkConfiguration = (image, configuration) => {
     path: '/system-connections/resin-wifi'
   }, wifiConfiguration.join('\n'))
 }
+
+module.exports = class ResinOS {
+  constructor (image, options = {}) {
+    this.image = image
```

Comment:
I don't think the constructor should pass the location here, but rather generate a temporary file name somewhere, and provide a getter method to obtain it. What do you think?